### DISCUSSION TITLE
Change pre-release versioning to adhere to semver properly

### DISCUSSION THREAD

**Discussion by siebertm** [activeadmin/activeadmin#8576] [General]
Hello!

I'm running into an issue when automatically upgrading `activeadmin` through [renovate](https://github.com/renovatebot/renovate). 

I'm running 

```ruby
#Gemfile
gem 'activeadmin', '4.0.0.beta13'
```
and

```jsonc
// package.json
{
  "dependencies": {
    "@activeadmin/activeadmin": "4.0.0.beta13"
  }
}
```

Now, when I want to have `renovate` upgrade this repository, I get the following upgrades:

* `activeadmin` gem to v4.0.0.beta15 (current at the time of writing this)
* `@activeadmin/activeadmin` npm to v4.0.0.beta9 (I'd expect v4.0.0.beta15!)

You might ask: tell the renovate folks! Well... Nay.

NPM wants us developers to strictly adhere to semver 2.0, which says about [prerelease identifiers](https://semver.org/#spec-item-9) that they should look like e.g. `v4.0.0.beta.1` (note the additional `.`)

Alphanumerical parts are to be [compared](https://semver.org/#spec-item-11) lexically, while numeric-only parts are compared numerically.

This means that `"alpha9" > "alpha11"` because of the lexical comparison. While `"alpha.9" < "alpha.11"`, as the parts are separately compared. 

My request would thus be to change pre-release versioning to adhere to semver 2.0 spec and thus release versions like `4.0.0.alpha.16`

*rubygems* does not enforce this and thus uses a more relaxed comparison, which should also work with the proposed scheme

**Comment by tagliala** [comment]
> NPM wants us developers to strictly adhere to semver 2.0, which says about [prerelease identifiers](https://semver.org/#spec-item-9) that they should look like e.g. v4.0.0.beta.1 (note the additional .)

Thanks, I was having this issue too, but I didn't figure out the cause.

Activeadmin is using the same approach as Rails, and it appears to not be a problem up to the 9th pre-release.

I would consider the change to the new versioning

**Comment by tagliala** [comment]
Example:

### package.json
```
    "@activeadmin/activeadmin": "4.0.0-beta14",
```

### yarn outdated
```
$ yarn outdated
yarn outdated v1.22.22
✨  Done in 1.54s.
```

### Expected

Suggest to update to .beta.15

### Actual

No outdated dependencies

---

I think this is too late for 4.0, but we should consider this for newer betas
